### PR TITLE
Nomenclature change

### PR DESCRIPTION
**Pull request by levibyte**

### CODE REVIEW

File: src/core/design.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "design.hpp"
+#include "core.hpp"
+
+IShapePtr Design::addObject(IShapePtr s)
+{
+	ASSERT_RETURN_VALUE(s != nullptr, nullptr);
+	auto obj = std::shared_ptr<IShape>(s->clone());
+	ASSERT_RETURN_VALUE(obj != nullptr, nullptr);
+    m_is_saved = false;
```
Comment (nagaina) [inline src/core/design.cpp]: alignment?

File: src/core/design.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	//	Interface
+	//
+	virtual ~Design() = default;
+
+public:
+    IShapePtr addObject(IShapePtr) override;
+	std::string getName() const noexcept override;
+
+public:
+    bool isSaved() { return m_is_saved; }
```
Comment (nagaina) [inline src/core/design.hpp]: consider using const, noexcept  where needed :P

File: src/core/iobject_pool.hpp [hunk 3]
```diff
@@ -23,15 +23,46 @@ class IObjectPool
...
+
+public:
+	//	Clear the content
+	void clear() noexcept override;
+
+	//	Content modiifiers
+	std::vector<IShapePtr> getObjects() const noexcept override;
+	IShapePtr addObject(IShapePtr) override;
+	void removeObject(IShapePtr) noexcept override;
+    virtual void dumpToFile(const std::string&) const override;
```
Comment (nagaina) [inline src/core/iobject_pool.hpp]: alignment

**Review by nagaina — CHANGES_REQUESTED** [review]

File: src/core/design.cpp (hunk 1 above)
Comment (levibyte) [reply src/core/design.cpp]: done

File: src/core/iobject_pool.hpp (hunk 3 above)
Comment (levibyte) [reply src/core/iobject_pool.hpp]: done

File: src/core/design.hpp (hunk 2 above)
Comment (levibyte) [reply src/core/design.hpp]: done

File: src/core/design.hpp [hunk 4]
```diff
@@ -17,10 +17,10 @@ class Design final : public ObjectPoolBase
 
 public:
     IShapePtr addObject(IShapePtr) override;
-	std::string getName() const noexcept override;
+    std::string getName() const noexcept override;
 
 public:
-    bool isSaved() { return m_is_saved; }
+    bool isSaved() const noexcept { return m_is_saved; }
```
Comment (nagaina) [inline src/core/design.hpp:23]: աբրիգոս ։Դ

**Review by nagaina — DISMISSED** [review]

**Pull request closed without merging**